src/daft-scan/src/glob.rs

```diff
@@ -36,55 +36,102 @@ fn run_glob(
-                let io_stats = IOStatsContext::new(format!(
-                    "GlobScanOperator constructor read_parquet_schema: for uri {first_filepath}"
-                ));
-                daft_parquet::read::read_parquet_schema(
-                    first_filepath,
-                    io_client,
-                    Some(io_stats),
-                    Default::default(), // TODO: pass-through schema inference options
-                )?
-            }
-            FileType::Csv => {
-                let io_client = get_io_client(true, io_config.clone())?; // it appears that read_parquet_schema is hardcoded to use multithreaded_io
-                let io_stats = IOStatsContext::new(format!(
-                    "GlobScanOperator constructor read_csv_schema: for uri {first_filepath}"
-                ));
-                let (schema, _, _, _, _) = daft_csv::metadata::read_csv_schema(
-                    first_filepath,
-                    true, // TODO: pass-through schema inference options
-                    None, // TODO: pass-through schema inference options
-                    true, // TODO: pass-through schema inference options
-                    None, // TODO: pass-through schema inference options
-                    io_client,
-                    Some(io_stats),
-                )?;
-                schema
+        let schema = match schema {
+            Some(s) => s,
+            None => {
+                let (io_runtime, io_client) = get_io_client_and_runtime(storage_config.as_ref())?;
+                let paths = run_glob(glob_path, Some(1), io_client.clone(), io_runtime)?;
+                let first_filepath = paths[0].as_str();
+                let inferred_schema = match file_format_config.as_ref() {
+                    FileFormatConfig::Parquet(ParquetSourceConfig {
+                        coerce_int96_timestamp_unit,
+                        row_groups: _,
+                    }) => {
+                        let io_stats = IOStatsContext::new(format!(
+                            "GlobScanOperator constructor read_parquet_schema: for uri {first_filepath}"
+                        ));
+                        daft_parquet::read::read_parquet_schema(
+                            first_filepath,
+                            io_client.clone(),
+                            Some(io_stats),
+                            ParquetSchemaInferenceOptions {
+                                coerce_int96_timestamp_unit: *coerce_int96_timestamp_unit,
+                            },
+                        )?
+                    }
+                    FileFormatConfig::Csv(CsvSourceConfig {
+                        delimiter,
+                        has_headers,
+                        double_quote,
+                        buffer_size: _,
+                        chunk_size: _,
```

Comment:
Can use the `..` syntax to ignored unused fields during pattern matching:
```suggestion
                        ..
```